### PR TITLE
Read aws launch config directly instead of reading from db

### PR DESCRIPTION
**Pull request by yujunglo**
@jinruh @sbaogang 
encoding user data and transforming role name into instance profile will be taken care by createLaunchConfigInternal (in AwsAutoScalingManager) and launchEC2Instances (in EC2HostInfoDAOImpl). 

decoding user data and transforming instance profile to role name will be take care by getLaunchConfigInfo (in AwsAutoScalingManager).

### CODE REVIEW

File: deploy-service/common/src/main/java/com/pinterest/arcee/aws/EC2HostInfoDAOImpl.java [hunk 1]
```diff
@@ -99,12 +106,14 @@ public void terminateHost(String hostId) throws Exception {
         RunInstancesRequest request = new RunInstancesRequest();
         request.setImageId(groupBean.getImage_id());
         request.setInstanceType(groupBean.getInstance_type());
-        request.setKeyName("ops");
+        request.setKeyName(vmKeyName);
         request.setSecurityGroupIds(Arrays.asList(groupBean.getSecurity_group()));
         request.setSubnetId(subnet);
-        request.setUserData(groupBean.getUser_data());
+        String userData = Base64.encodeBase64String(groupBean.getUser_data().getBytes());
```
Comment (jinruh) [inline deploy-service/common/src/main/java/com/pinterest/arcee/aws/EC2HostInfoDAOImpl.java]: Just to double check: the userdata we read from groupbean is always decoded string, right?

Comment (yujunglo) [inline deploy-service/common/src/main/java/com/pinterest/arcee/aws/EC2HostInfoDAOImpl.java]: Yes, groupBean always stores decoding data.

File: deploy-service/common/src/main/java/com/pinterest/arcee/autoscaling/AwsAutoScalingManager.java [hunk 2]
```diff
@@ -548,25 +547,43 @@ public void enableScalingDownEvent(String groupName) throws Exception {
...
-        if (group == null) {
-            LOG.warn(String.format("Failed to get cluster %s: auto scaling group %s does not exist", groupName, groupName));
+    public AwsVmBean getAutoScalingGroupInfo(String clusterName) throws Exception {
+        AutoScalingGroup autoScalingGroup = getAutoScalingGroup(clusterName);
+        if (autoScalingGroup == null) {
+            LOG.warn(String.format("Failed to get cluster %s: auto scaling group %s does not exist", clusterName, clusterName));
             return null;
         }
 
+        AwsVmBean launchConfigInfo = getLaunchConfigInfo(clusterName, autoScalingGroup.getLaunchConfigurationName());
```
Comment (jinruh) [inline deploy-service/common/src/main/java/com/pinterest/arcee/autoscaling/AwsAutoScalingManager.java]: In theory, you probably don't need the clusterName as parameter to the API.

**Comment by yujunglo** [comment]
updaed

**Comment by jinruh** [comment]
r+ thanks! looks good!

**Pull request merged**